src/server/replica.cc

```diff
@@ -922,6 +926,18 @@ void Replica::StableSyncDflyReadFb(Context* cntx) {
       ExecuteTxWithNoShardSync(std::move(*tx_data), cntx);
     }
 
+    // Handle ACKs with the master. PING commands from the master mean we should immediately
+    // answer.
+    uint64_t current_offset = journal_rec_executed_.load(std::memory_order_relaxed);
+    bool force_ack =
+        !tx_data->commands.empty() && ToSV(tx_data->commands.front().cmd_args[0]) == "PING";
+    if (force_ack || current_offset > ack_offs_ + 1024 || time(nullptr) > last_ack + 1) {
+      ack_cmd = absl::StrCat("REPLCONF ACK ", current_offset);
+      last_ack = time(nullptr);
+      ack_offs_ = current_offset;
+      CHECK(!SendCommand(ack_cmd, &serializer));
+    }
+
```

Comment:
Not sure how to comment its still a draft. Review on draft for me mostly means: check my approach, but don't dig into code details/error handling/comments etc

1. Maybe we should send it in a separate fiber to avoid blocking? What should we do if it failed?
2. Lets move it to a separate function, because its getting big